### PR TITLE
Added support for Noclone and Optimize Function Attribute

### PR DESCRIPTION
**Pull request by shubhamnarlawar77**

### CODE REVIEW

**Comment by shubhamnarlawar77** [comment]
'@marxin : review request'

File: src/Function.cpp [hunk 1]
```diff
@@ -160,6 +160,7 @@ Function::GenerateAttributes()
 
 		attributes.push_back(new MultiValuedFunctionAttribute("visibility", FuncAttrProb, {"default", "hidden", "protected", "internal"}));
 		attributes.push_back(new MultiValuedFunctionAttribute("no_sanitize", FuncAttrProb, {"address", "thread", "undefined", "kernel-address", "pointer-compare", "pointer-subtract", "leak"}));
+		attributes.push_back(new MultiValuedFunctionAttribute("optimize", FuncAttrProb, {"-O0", "-O1", "-O2", "-O3", "-Os", "-Ofast"}));
```
Comment (marxin) [inline src/Function.cpp]: Please include also `-Og`.

**Review by marxin — CHANGES_REQUESTED** [review]
Just one note, merge it after that change.

File: src/Function.cpp (hunk 1 above)
Comment (shubhamnarlawar77) [reply src/Function.cpp]: Done.

**Comment by marxin** [comment]
The patch is fine, thanks.

**Comment by shubhamnarlawar77** [comment]
> The patch is fine, thanks.

I will send type attributes patch after this merge as discussed in mail. Is this fine?

**Comment by marxin** [comment]
Yep.

**Pull request merged**